pkg/controllers/git/git.go

```diff
@@ -97,6 +97,18 @@ func targetsOrDefault(targets []fleet.GitTarget) []fleet.GitTarget {
 	return targets
 }
 
+func targetRestrictionsOrDefault(targets []fleet.GitTargetRestriction) []fleet.GitTargetRestriction {
+	if len(targets) == 0 {
+		return []fleet.GitTargetRestriction{
+			{
+				Name:         "default",
+				ClusterGroup: "default",
+			},
+		}
```

Comment:
Does this replicate the default behavior today from your testing?

Reply:
tbh i have not been able to figure out a good way to test this. unless i missed some documentation, its not clear how to effectively set up a testing env.

Reply:
Testing ENVs are usually set up by installing a local cluster of some sort (k3d/k3s) and then installing the fleet charts directly. Many people install a local cluster, then install the latest Rancher v2.5.x release to get Fleet in the GUI. With both approaches though, there is only a local cluster (unless you create a downstream cluster).